interp/interp_test.go

```diff
@@ -2982,6 +2982,27 @@ done <<< 2`,
 		"mapfile -t butter <<EOF\na\nb\nc\nEOF\n" + `for x in "${butter[@]}"; do echo "$x"; done`,
 		"a\nb\nc\n",
 	},
+	// read & EOF
+	// // newline
+	// {
+	// 	`a=a; echo | read a; echo -n "$a"`,
+	// 	"",
+	// },
+	// empty input
+	{
+		`a=b; read a < /dev/null; echo -n "$a"`,
+		"",
+	},
+	// // some string and newline
+	// {
+	// 	"a=c; echo x | read a; echo -n $a",
+	// 	"x",
+	// },
+	// // some string and EOF
+	// {
+	// 	"a=d; echo -n y | read a; echo -n $a",
+	// 	"y",
+	// },
```

Comment:
did you mean to leave most of these commented out?

Reply:
Yes. Sort of. I mentioned in the commit:

- Add some test cases. Daniel suggested four, three of which already
  pass without this PR. I left them in but commented-out, in case
  somebody wants to look at them.

Should uncomment them or remove them entirely? Wasn't sure how you'd feel about that either way.

Reply:
Sorry, maybe I'm being slow as it's a bit late. If these test cases already behaved the way we want in master, why not enable them as regression tests, even if they're not directly related to this fix?

Reply:
Sure, I'm fine with that.